WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -363,11 +388,34 @@ public int compare(SiteRecord site1, SiteRecord site2) {
         @Override
         protected void onPostExecute(SiteList sites) {
             if (mSites == null || !mSites.isSameList(sites)) {
-                mSites = sites;
+                mAllSites = (SiteList) sites.clone();
+                mSites = filteredSitesByTextIfInSearchMode(sites);
```

Comment:
It would be preferable to do the filtering in `doInBackground` rather than in the main thread here.